src/topology/mod.rs

```diff
@@ -31,6 +34,8 @@ type TaskHandle = tokio::task::JoinHandle<Result<(), ()>>;
 
 #[allow(dead_code)]
 pub struct RunningTopology {
+    #[cfg(feature = "api")]
+    api: Option<Server>,
```

Comment:
I would really prefer pretty much all of the changes in this file not happen :smile: (other than the spelling fixes, which should get their own PR).

We're currently working to try to simplify the topology code as much as possible and I just don't see any real benefit to putting it in change of the API server. We should start with something as simple as possible and just spawn the server in `main` when it's configured.

Reply:
I currently lack a lot of context about what's stable or subject to change, and still fumbling my way through the code base, so bear with my lack of understanding here... but if the topology isn't updated, I'm not clear how we'd respond to config changes.

If someone updates:

```toml
[api]
  bind = "<new_ip_port>
```

... we'd want to respond to that the same as any other config, right?